plugins/pulp_rpm/plugins/migrations/0037_rpm_primary_repodata_new_location.py

```diff
@@ -0,0 +1,171 @@
+from cStringIO import StringIO
+import sys
+import os
+
+from pulp.server.db import connection
+
+if sys.version < (2, 7):
+    # We need the non-C implementation so we can shove a namespace into its internal data structure,
+    # since in python 2.6, the register_namespace function wasn't present.
+    import xml.etree.ElementTree as ET
+else:
+    import xml.etree.cElementTree as ET
+
+
+RPM_NAMESPACE = 'http://linux.duke.edu/metadata/rpm'
+PULP_PACKAGES_DIR = "Packages"
+
+# this is required because some of the pre-migration XML tags use the "rpm"
+# namespace, which causes a parse error if that namespace isn't declared.
+FAKE_XML = '<?xml version="1.0" encoding="%(encoding)s"?><faketag ' \
+           'xmlns:rpm="%(namespace)s">%(xml)s</faketag>'
+
+
+def migrate(*args, **kwargs):
+    """
+    Migrate RPMs and SRPMs location to new format "Packages/[a-z]/.*".
+
+    Remove in DRPMs "drpms/" prefix in relative_path.
+
+    :param args:   unused
+    :type  args:   list
+    :param kwargs: unused
+    :type  kwargs: dict
+    """
+
+    db = connection.get_database()
+    rpm_collection = db['units_rpm']
+    srpm_collection = db['units_srpm']
+    drpm_collection = db['units_drpm']
+
+    for rpm in rpm_collection.find({}, ['repodata', 'filename']):
+        migrate_rpm_base(rpm_collection, rpm)
+    for srpm in srpm_collection.find({}, ['repodata', 'filename']):
+        migrate_rpm_base(srpm_collection, srpm)
+    for drpm in drpm_collection.find({}, ['filename']):
+        migrate_drpm(drpm_collection, drpm)
+
+
+def migrate_drpm(collection, unit):
+    """
+    Migrate DRPMs filename field to new format.
```

Comment:
Interesting... I somewhat agree with both of you :)
I do not think we want to do this but indeed the filename field contains data like `drpms/foo_1.1_1.2.drpm` which is confusing. Was it the way to be sure to have drpms in a separate directory?

If you'll take a look at [the code](https://github.com/pulp/pulp_rpm/blob/2.11-dev/plugins/pulp_rpm/plugins/distributors/yum/publish.py#L562-L563) it extracts filename (so no use of `drpms/` from filename field) and then adds `drpms/` back. 
So I suggest to leave filename field as-is in db in case this `drpms/` part is used somewhere and to be more confident that we do not introduce regression here. We generate the correct path during publish anyway, both in prestodelta generation and symlink creation.
What do you think? @BrnoPCmaniak  @mhrivnak